pkg/storage/storage.go

```diff
@@ -314,6 +314,47 @@ func (s *Storage) Get(gi *GetInput) (*GetOutput, error) {
 	}, nil
 }
 
+type CleanupInput struct {
+	Key            *Key
+	DepthThreshold int
+	TimeThreshold  time.Time
+}
+
+func (s *Storage) Cleanup(ci *CleanupInput) error {
+	s.closingMutex.Lock()
+	defer s.closingMutex.Unlock()
+
+	if s.closing {
+		return errClosing
+	}
+
+	lg := logrus.WithField("key", ci.Key.Normalized())
+
+	sk := ci.Key.SegmentKey()
+	st := s.segments.Get(sk).(*segment.Segment)
+	st.Get(ci.TimeThreshold, ci.TimeThreshold, func(depth int, samples, writes uint64, t time.Time, r *big.Rat) {
```

Comment:
Couple of problems with this approach:
* `Get` will only return a limited number of nodes, because it optimizes for the smallest number of nodes to return as possible
```
|__________________________|
|XXXXXXXX|________|________|
|__|__|__|XX|XX|__|__|__|__|
    ^          ^-threshold
    missed nodes
```
* We should also delete the relevant nodes from the segment tree, not just `s.trees`

Reply:
I think the way to solve this is to write a dedicated `Cleanup` function for the segment tree as well.